File: Examples/Mvc.CS/Controllers/DocumentViewerController.Events.cs
using System.Web.Mvc;
using GleamTech.DocumentUltimate.AspNet.UI;
using GleamTech.Examples;

namespace GleamTech.DocumentUltimateExamples.Mvc.CS.Controllers
{
    public partial class DocumentViewerController
    {
        [AcceptVerbs(HttpVerbs.Get | HttpVerbs.Post)]
        public ActionResult Events()
        {
            var exampleFileSelector = ViewBag.ExampleFileSelector = new ExampleFileSelector
            {
                Id = "exampleFileSelector",
                InitialFile = "Default.pdf"
            };

            var documentViewer = new DocumentViewer
            {
                Width = 800,
                Height = 600,
                Resizable = true,
                Document = exampleFileSelector.SelectedFile,
                ClientLoad = "documentViewerLoad",
                ClientError = "documentViewerError",
                ClientDocumentLoad = "documentViewerDocumentLoad",
                ClientPageChange = "documentViewerPageChange",
                ClientPageComplete = "documentViewerPageComplete",
                ClientRotationChange = "documentViewerRotationChange",
                ClientPrint = "documentViewerPrint",
                ClientDownload = "documentViewerDownload",
                ClientDownloadAsPdf = "documentViewerDownloadAsPdf",
                ClientPrintStart = "documentViewerPrintStart",
                ClientPrintProgress = "documentViewerPrintProgress"
            };

            return View(documentViewer);
        }
    }
}

File: Examples/Mvc.VB/Controllers/DocumentViewerController.Events.vb
Imports GleamTech.DocumentUltimate.Web
Imports GleamTech.ExamplesCore

Namespace Controllers
    Partial Public Class  DocumentViewerController
        Inherits Controller

        <AcceptVerbs(HttpVerbs.Get Or HttpVerbs.Post)>
	    Public Function Events(fileSelector As String) As ActionResult
            Dim exampleFileSelector =  new ExampleFileSelector() With {
                .ID = "exampleFileSelector",
                .InitialFile = "PDF Document.pdf"
            }
            ViewBag.ExampleFileSelector = exampleFileSelector

		    Dim documentViewer = New DocumentViewer() With { 
			    .Width = 800, 
			    .Height = 600,
                .Resizable = True,
			    .Document = exampleFileSelector.SelectedFile,
		        .ClientLoad = "documentViewerLoad",
		        .ClientError = "documentViewerError",
		        .ClientDocumentLoad = "documentViewerDocumentLoad",
		        .ClientPageChange = "documentViewerPageChange",
		        .ClientPageComplete = "documentViewerPageComplete",
		        .ClientPrint = "documentViewerPrint",
		        .ClientDownload = "documentViewerDownload",
		        .ClientDownloadAsPdf = "documentViewerDownloadAsPdf",
		        .ClientPrintStart = "documentViewerPrintStart",
		        .ClientPrintProgress = "documentViewerPrintProgress"
		    }

		    Return View(documentViewer)
	    End Function

    End Class
End Namespace
